darwin/DarwinProcess.c

```diff
@@ -240,6 +241,12 @@ void DarwinProcess_setFromKInfoProc(Process* proc, const struct kinfo_proc* ps,
    proc->updated = true;
 }
 
+static double ticksToNanoseconds(const double ticks) {
+   const double secs_per_tick = 1.0 / (double) Platform_clockTicksPerSec;
+   const double nanos_per_sec = 1e9;
+   return ticks * secs_per_tick * nanos_per_sec;
```

Comment:
For better precision avoid multiplications with small numbers; instead prefer `ticks * nanos_per_sec / Platform_clockTicksPerSec`. Overall saves on one multiplication and thus reduces the round-off error introduced in these operations.

Reply:
My $0.02, I'm more concerned about correctness/clarity than saving a few hundred multiplications.

Precision isn't much of a concern here, IMO:
1) `1.0 / 100.0 * 1e9` is losslessly representable/computable with `double`. It could even be done with `float` actually, and still wouldn't lose precision when `Platform_clockTicksPerSec == 100`
2) Large numbers are also an issue for precision. Doubles start truncating integer representations to even numbers after `9,007,199,254,740,993 (253 + 1).`. The computation of `ticks * nanos_per_sec` could become a pretty large value itself, which threatens precision. The issue would kick in after `9,007,199` ticks. At 100 ticks/s, that's 104 days of uptime with 1 CPU core, 52 days on a dual core, 26 days on a quad core, ... and so on. A maximally precise computation would be alternating multiplications and divisions in a way that stays in the more precise regions of `double`. So something like  `ticks / Platform_clockTicksPerSec * nanos_per_sec`
3) And more fundamentally, this computation is inescapably imprecise, since the process' times are updated non-atomically, separately from the elapsed CPU ticks. There will be a variable time between the measurements, that introduces inaccuracy.
4) None of this really matters anyway, since the result is only presented to 3 significant digits.

I don't mind changing it to whatever your suggest, but I would advise prioritizing clarity.

Reply:
The suggestion from 2 seems fine with me.

FWIW: Neither `0.1` nor `0.01` are representable exactly in neither `float` nor `double`. Thus if there are simple ways to avoid predictable imprecision you oughta go them.